src/actions/utils.cr

```diff
@@ -0,0 +1,76 @@
+module DppmRestApi::Actions
+  # A series of utility functions that are useful to all of the Action blocks
+  module Utils
+    macro fmt_route(route = "", namespace = false)
+    {{ "/" + @type.stringify
+         .downcase
+         .gsub(/^DppmRestApi::Actions::/, "")
+         .gsub(/::/, "/") }} {% if namespace %} + "/:namespace" {% end %} + ( {{route}} || "")
+    end
+
+    # the "last" or -1st block argument is selected because context is always the
+    # argument -- either the only or the second of two
+    {% for method in %w(get post put delete ws options) %}
+    macro relative_{{method.id}}(route, &block)
+      {{method.id}} fmt_route(\{{route}}) do |\{{block.args.splat}}|
+        namespace = \{{block.args[-1]}}.params.query["namespace"]? || DPPM::Prefix.default_group
+        \{{block.body}}
+      end
+    end
+    {% end %}
+
+    # Pull the context from the context's query parameter or use the default
+    # provided by the CLI.
+    #
+    # This macro only works inside of a route block with the variable "context"
+    # available
+    def get_prefix_or_default(from context)
+      DPPM::Prefix.new context.params.query["prefix"]? || DPPM.default_prefix
+    end
+
+    # A common mistake for users of HTTP APIs is to when working with boolean
+    # query parameters (I.E. `?value=false` is actually truthy). This method
+    # attempts to guard against that mistake by making `false`, `0`and `no`
+    # (case insensitive) falsey for our purposes. That is, this method returns
+    # `false` if the key is not present, or in the case of something like
+    # `?write=0`. In any other case it returns `true`.
+    #
+    # Anyone who is aware of this common mistake will of course instead use
+    # the proper boolean format, like `?value&other`
+    def parse_boolean_param(key : String, from context : HTTP::Server::Context) : Bool
+      if param = context.params.query[key]?
+        return false unless ["false", "no", "0"].includes? param.downcase
```

Comment:
```suggestion
        return false unless {"false", "no", "0"}.includes? param.downcase
```
Why we also support `yes` and `no`?
Can you provide some documentation for this like MDN?

Reply:
I can remove "no", I agree that that's silly.

Reply:
Also I don't know where I got the idea that boolean parameters were standardized to be `?value=false` being truthy. That seems to just be wrong.